packages/pipeline/migrations/1548809952793-AllowDuplicateTxHashesInDexTrades.ts

```diff
@@ -0,0 +1,27 @@
+import { MigrationInterface, QueryRunner, TableColumn } from 'typeorm';
+
+const DEX_TRADES_TABLE_NAME = 'raw.dex_trades';
+
+export class AllowDuplicateTxHashesInDexTrades1548809952793 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<any> {
+        const dexTradesTable = await queryRunner.getTable(DEX_TRADES_TABLE_NAME);
+        if (dexTradesTable) {
+            // Composite key goes from (source_url, tx_hash) to (trade_index, source_url, tx_hash)
+            await queryRunner.addColumn(
+                DEX_TRADES_TABLE_NAME,
+                new TableColumn({
+                    name: 'trade_index',
```

Comment:
I think we should prefix this as `bloxy_trade_index` instead.

Also worried about what happens if we want to use other sources that don't have a trade index. Maybe we should start thinking of this table as containing bloxy info only. The primary key constraint means this column can't be null.

Reply:
I feel like a `trade_index`-like thing will be relevant for every source. We just need (source_url, tx_hash, trade_index) to be unique. If we start getting trades from other sources, we will also need `trade_index` for uniqueness.